nucypher/characters/base.py

```diff
@@ -366,121 +357,34 @@ def encrypt_for(self,
-        :param sign_plaintext: When signing, the cleartext is signed if this is
+        :param sign_plaintext: the cleartext is signed if this is
             True,  Otherwise, the resulting ciphertext is signed.
 
-        :return: A tuple, (ciphertext, signature).  If sign==False,
-            then signature will be NOT_SIGNED.
+        :return: the message kit.
         """
-        signer = self.stamp if sign else DO_NOT_SIGN
+
+        # TODO: who even uses this method except for tests?
 
         message_kit = MessageKit.author(recipient_key=recipient.public_keys(DecryptingPower),
                                         plaintext=plaintext,
-                                        signer=signer,
+                                        signer=self.stamp.as_umbral_signer(),
                                         sign_plaintext=sign_plaintext)
         return message_kit
 
-    def verify_from(self,
-                    stranger: 'Character',
-                    message_kit: Union[MessageKit, PolicyMessageKit, bytes],
-                    signature: Signature = None,
-                    decrypt=False,
-                    label=None,
-                    ) -> bytes:
-        """
-        Inverse of encrypt_for.
-
-        :param stranger: A Character instance representing
-            the actor whom the sender claims to be.  We check the public key
-            owned by this Character instance to verify.
-        :param message_kit: the message to be (perhaps decrypted and) verified.
-        :param signature: The signature to check.
-        :param decrypt: Whether or not to decrypt the messages.
-        :param label: A label used for decrypting messages encrypted under its associated policy encrypting key
-
-        :return: Whether or not the signature is valid, the decrypted plaintext or NO_DECRYPTION_PERFORMED
-        """
-
-        sender_verifying_key = stranger.stamp.as_umbral_pubkey()
-
-        # We're duck-typed here - we want to accept a message kit object or bytes.
-        # This gives us a fail fast opportunity:
-        # If the higher-order object MessageKit is passed, we can perform an additional
-        # eager sanity check before performing decryption.
-
-        with contextlib.suppress(AttributeError):
-            if message_kit.sender_verifying_key:
-                if not message_kit.sender_verifying_key == sender_verifying_key:
-                    raise ValueError("This MessageKit doesn't appear to have come from {}".format(stranger))
-
-        #
-        # Decrypt
-        #
-
-        signature_from_kit = None
-        if decrypt:
-
-            # We are decrypting the message; let's do that first and see what the sig header says.
-            cleartext_with_sig_header = self.decrypt(message_kit=message_kit, label=label)
-            sig_header, cleartext = default_constant_splitter(cleartext_with_sig_header, return_remainder=True)
+    def decrypt_internal(self,
```

Comment:
What's the reason for us calling this `decrypt_internal()` - what is `internal` in the name attempting to convey? A little weird having a public function named with `internal`